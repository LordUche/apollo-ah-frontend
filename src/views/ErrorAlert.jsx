import React from 'react';
import propTypes from 'prop-types';

/**
 * @class ErrorAlert
 * @extends { React }
 */
class ErrorAlert extends React.Component {
  render() {
    const { message, close } = this.props;
    if (!message) {
      return (
        ''
      );
    }
    return (
      <div className="alert alert-danger alert-dismissible fade show" role="alert">
        <strong>Error!    </strong>
        { message.message }
        <button type="button" className="close" data-dismiss="alert" aria-label="Close" onClick={close}>
          <span aria-hidden="true">&times;</span>
        </button>
      </div>
    );
  }
}

ErrorAlert.propTypes = {
  message: propTypes.object,
  close: propTypes.func
};

export default ErrorAlert;
