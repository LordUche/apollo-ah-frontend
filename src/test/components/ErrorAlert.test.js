import React from 'react';
import { shallow } from 'enzyme';
import ErrorAlert from '../../views/ErrorAlert';

const props = {
  message: { message: 'Email is required', field: 'email' },
  close: () => true,
};

describe('<ErrorAlert Test Suite>', () => {
  describe('<ErrorAlert>', () => {
    it('It should render succesfully', () => {
      const wrapper = shallow(<ErrorAlert {...props} />);
      expect(wrapper).toBeDefined();
      expect(wrapper).toMatchSnapshot();
    });
  });
});
