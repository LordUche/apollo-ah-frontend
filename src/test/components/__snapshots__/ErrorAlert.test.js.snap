// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorAlert Test Suite> <ErrorAlert> It should render succesfully 1`] = `
<div
  className="alert alert-danger alert-dismissible fade show"
  role="alert"
>
  <strong>
    Error!    
  </strong>
  Email is required
  <button
    aria-label="Close"
    className="close"
    data-dismiss="alert"
    onClick={[Function]}
    type="button"
  >
    <span
      aria-hidden="true"
    >
      Ã—
    </span>
  </button>
</div>
`;
